File: 프로그래머스/unrated/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.java
class Solution {
    public int[] solution(int[] arr, int[][] q) {
        int ans[] = new int[q.length];
        
        for (int i = 0; i < q.length; i++)
        {
            ans[i] = -1;
            for (int j = q[i][0]; j <= q[i][1]; j++)
            {
                if (q[i][2] < arr[j] && (arr[j] < ans[i] || ans[i] == -1))
                        ans[i] = arr[j];
            }
        }
        return ans;
    }
}

File: 프로그래머스/Lv.0/181923. 수열과 구간 쿼리 2/수열과 구간 쿼리 2.js
function solution(arr, queries) {
    var answer = [];
    for (const q of queries) {
        var min = Number.MAX_VALUE;
        for (var i = q[0]; i <= q[1]; i++)
            if (q[2] < arr[i] && arr[i] < min)
                min = arr[i];
        answer.push(min === Number.MAX_VALUE ? -1 : min);        
    }
    return answer;
}